### PR TITLE
adding ignore file

### PR DESCRIPTION
**Pull request by JohnLieske**
This is introducing an ignore file (this one is specific to 2.12 as that's the latest) with entries to ignore import & compile for python 2.6 in the testing suite.

### CODE REVIEW

**Comment by JohnLieske** [comment]
This was meant to be two PRs, but both commits went at the same time.

The second commit was corrections for the pep8 & some other errors in the following files:

plugins/modules/ipaautomember.py
plugins/modules/ipaconfig.py
plugins/modules/ipadnsconfig.py
plugins/modules/ipadnsrecord.py
plugins/modules/ipagroup.py
plugins/modules/iparole.py
plugins/modules/ipaservice.py
plugins/modules/ipauser.py
plugins/modules/ipavault.py
plugins/module_utils/ansible_freeipa_module.py

**Comment by t-woerner** [comment]
Thanks for the PR.
There is an issue though introduced by this PR.

File: plugins/modules/ipaautomember.py [hunk 1]
```diff
@@ -153,7 +153,7 @@ def gen_condition_args(grouping,
     _args = {}
     if grouping is not None:
         _args['type'] = grouping
-    if key is not None:
+    if key is not None:2.12
```
Comment (t-woerner) [inline plugins/modules/ipaautomember.py:156]: This needs to be fixed.

Comment (JohnLieske) [reply plugins/modules/ipaautomember.py]: Apologies, looks like that was accidentally added.

File: plugins/module_utils/ansible_freeipa_module.py [hunk 2]
```diff
@@ -108,7 +108,9 @@ def parse(version_str):
     try:
         from collections.abc import Mapping  # noqa
     except ImportError:
+        #pylint: disable=wrong-import-position
```
Comment (rjeffman) [inline plugins/module_utils/ansible_freeipa_module.py:111]: As `wrong-import-position` is a regular issue also with plugins, can't it be disabled globally? This is what we currently do in `ansible-freeipa`.

File: plugins/modules/ipaconfig.py [hunk 3]
```diff
@@ -418,12 +418,12 @@ def main():
                 if ansible_module.argument_spec.get(k):
                     arg_type = ansible_module.argument_spec[k]['type']
                     if k in (
-                           'ipaselinuxusermaporder', 'domain_resolution_order'
-                       ):
+                        'ipaselinuxusermaporder', 'domain_resolution_order'
+                        ):
```
Comment (rjeffman) [inline plugins/modules/ipaconfig.py:422]: This format is both weird, and makes flake8 complains about the line below the test condition starting at the same indentation (also, it makes code harder to read, in my opinion).
What do you think about something like:
```python
if (
    k in ('some', 'options')
):
    # here goes the code block.
```
or if 'k in ()' being too long:
```python
if (
    k in (
        "some",
        "really long",
        "options",
    )
):
```

File: plugins/modules/ipadnsrecord.py [hunk 4]
```diff
@@ -1110,7 +1110,7 @@ def configure_module():
 
             records=dict(type="list", default=None,
                          options=dict(
-                            # Here name is a simple string
+                            #  Here name is a simple string
```
Comment (rjeffman) [inline plugins/modules/ipadnsrecord.py:1113]: This shouldn't be an issue, is it?

File: plugins/modules/ipagroup.py [hunk 5]
```diff
@@ -395,11 +395,11 @@ def main():
                         # or nonposix to external.
                         if not compare_args_ipa(ansible_module, args,
                                                 res_find) or \
-                           (
+                            (
                                not is_posix_group(res_find) and
                                not is_external_group(res_find) and
                                (posix or external)
-                           ):
+                            ):
```
Comment (rjeffman) [inline plugins/modules/ipagroup.py:402]: Same issue as in `plugins/modules/ipaconfig.py` line 422.

**Review by rjeffman — COMMENTED** [review]
Thank you very much for the PR. I have some comments about few changes.

Do we have a sample configuration for the pylint linter used so that we can compare to the configuration we currently use?

I wonder why our current linter configuration didnt' get a few of this problems.

File: plugins/modules/ipaconfig.py (hunk 3 above)
Comment (rjeffman) [reply plugins/modules/ipaconfig.py:422]: Yes, this is somewhat "black-like" formatting, but this is a hard one. ;-)

**Comment by JohnLieske** [comment]
I can ask about a sample configuration for the linter (I don't have that answer right now). The indenting is showing up for us (when we download the tarball & check) with both spaces & indents, which is most likely why it's kicking out the errors (per pep8 indent standards: https://www.python.org/dev/peps/pep-0008/ ).

I can ask & get back to you about a linting config if you require it.

**Comment by rjeffman** [comment]
@JohnLieske the identation error is puzzling me, as we do have linter checks (both pylint and flake8) on CI and pre-commit. It weird that it is happening.

@t-woerner, we (as in I) shuld check if the build script is not mangling with spaces.

**Comment by rjeffman** [comment]
@JohnLieske, thank you for this PR.

This changes have been incorporated by #709, so I'm closing this one.

**Pull request closed without merging**